plugins/openweather/helpers.rb

```diff
@@ -0,0 +1,72 @@
+module AresMUSH
+    module Openweather
+      mattr_accessor :current_weather
+  
+      def self.can_change_weather?(actor)
+        actor.has_permission?("manage_weather")
+      end
+  
+      def self.load_weather_if_needed
+        if (Openweather.current_weather == {})
+          Openweather.change_all_weathers
+        end
+      end
+  
+      def self.change_all_weathers
+        # Set an initial weather for each area and the default one
+        Openweather.current_weather = {}
```

Comment:
Since this is using an external API, you don't want to block your main game thread for N area queries.  I would suggest making this a background thread call and putting `change_weather(area)` inline (so it can't be called outside of the background task) 

```
        # nil is for client, which you don't have here.
        Global.dispatcher.spawn("Querying weather", nil) do
            areas.each do |a|
              ... stuff from change_weather(area) ...
            end
        end
       end
```

Reply:
Thank you! Made this change

Reply:
This made a big performance change. Thank you